From d36bdd3acd72c6a0abdbe4c6236af82f68796acd Mon Sep 17 00:00:00 2001
From: mbegumx <mogalx.begum@intel.com>
Date: Fri, 23 Oct 2020 15:19:03 +0530
Subject: [PATCH] Support for 12 CPU cores

    Updating the maximum cpu cores to get cpu usage.
    Fix vts issue in CpuUsageTest

Tracked-On: OAM-94409
Signed-off-by: mbegumx <mogalx.begum@intel.com>

diff --git a/thermal/Thermal.cpp b/thermal/Thermal.cpp
index 8fcad6c..5fd1a8b 100644
--- a/thermal/Thermal.cpp
+++ b/thermal/Thermal.cpp
@@ -25,7 +25,7 @@
 #include "Thermal.h"

 #define SYSFS_TEMPERATURE_CPU       "/sys/class/thermal/thermal_zone0/temp"
-#define CPU_NUM_MAX                 4
+#define CPU_NUM_MAX                 12
 #define CPU_USAGE_PARAS_NUM         5
 #define CPU_USAGE_FILE              "/proc/stat"
 #define CPU_ONLINE_FILE             "/sys/devices/system/cpu/online"
@@ -44,7 +44,18 @@ using ::android::hardware::interfacesEqual;
 using ::android::hardware::thermal::V1_0::ThermalStatus;
 using ::android::hardware::thermal::V1_0::ThermalStatusCode;

-static const char *CPU_LABEL[] = {"CPU0", "CPU1", "CPU2", "CPU3"};
+static const char *CPU_LABEL[] = {"CPU0",
+                                  "CPU1",
+                                  "CPU2",
+                                  "CPU3",
+                                  "CPU4",
+                                  "CPU5",
+                                  "CPU6",
+                                  "CPU7",
+                                  "CPU8",
+                                  "CPU9",
+                                  "CPU10",
+                                  "CPU11"};
 struct zone_info {
 	uint32_t temperature;
 	uint32_t trip_0;
2.29.0

